src/main/java/uk/gov/justice/hmpps/casenotes/config/SecurityUserContext.java

```diff
@@ -32,11 +27,23 @@ private Authentication getAuthentication() {
         return SecurityContextHolder.getContext().getAuthentication();
     }
 
-    public String getCurrentUsername() {
-        final String username;
+    public Optional<String> getCurrentUsername() {
+        return getOptionalCurrentUser().map(User::getUsername);
+    }
+
+    public UserIdUser getCurrentUser() {
+        return getOptionalCurrentUser().orElseThrow(() -> new IllegalStateException("Current user not set but is required"));
+    }
 
-        final var userPrincipal = getUserPrincipal();
+    private Optional<UserIdUser> getOptionalCurrentUser() {
+        final var authentication = getAuthentication();
+        if (authentication == null || authentication.getPrincipal() == null) return Optional.empty();
 
+        final var userPrincipal = authentication.getPrincipal();
```

Comment:
Very very minor comment - might be cleaner to move this line ^^ above the null check and replace authentication.getPrincipal() == null with userPrincipal == null to be more consistent.